.github/scripts/sink_cache.py

```diff
@@ -0,0 +1,235 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+# CLI to Delete Github Action Cache.
+#
+# This script remove Cache entries a specified Github repository.
+# This allow future workflow to run on clear cache data (since they'll have to recreate them)
+import argparse
+from datetime import datetime
+from functools import reduce
+import json
+import logging
+import subprocess
+from typing import Any, Dict, Optional
+from logging import basicConfig, getLogger
+
+import sys
+from os.path import dirname, abspath
+
+current_dir = dirname(abspath(__file__))
+parent_dir = dirname(dirname(current_dir))
+sys.path.insert(0, parent_dir)
+
+import misc.bench as bench
+
+basicConfig(level=logging.INFO)
+logger = getLogger()
+
+
+def gh_api(url: str, method: Optional[str] = "GET"):
+    extra_args = [f"--header='Accept: application/json'"]
+    if method is not None:
+        extra_args.append(f"--method={method}")
+    cmd_args = ["gh", "api", *extra_args, url]
+    logger.debug(f"Will execute cmd `{' '.join(cmd_args)}`")
+    ret = subprocess.check_output(cmd_args)
+    return json.loads(ret)
+
+
+class CacheEntry:
+    id: int
+    key: str
+    ref: str
+    version: str
+    last_accessed_at: datetime
+    created_at: datetime
+    size_in_bytes: int
+
+    def __init__(
+        self,
+        id: int,
+        key: str,
+        ref: str,
+        version: str,
+        last_accessed_at: datetime,
+        created_at: datetime,
+        size_in_bytes: int,
+    ) -> None:
+        self.id = id
+        self.key = key
+        self.ref = ref
+        self.version = version
+        self.last_accessed_at = last_accessed_at
+        self.created_at = created_at
+        self.size_in_bytes = size_in_bytes
+
+    def from_dict(dict: Dict[str, Any]) -> "CacheEntries":
+        # Datetime format for `2019-01-24T22:45:36.000Z``
+
+        def get_iso_datetime(raw: str) -> datetime:
+            # `fromisoformat` don't support:
```

Comment:
Not sure why you need to handle nanosecond fallback, [according to github documentation](https://docs.github.com/en/rest/actions/cache#list-github-actions-caches-for-a-repository) they only return up to the millisecond (which is the most common for a web service)

So you should only have to do:
```python
dt = datetime.fromisoformat(d.replace("Z", "+00:00"))  # `fromisoformat` doesn't support Z suffix
```

Reply:
The actual data return a datetime with the nanosecond 🤷

Reply:
I've rework `get_iso_datetime`